Editor/UI/Actions.cs

```diff
@@ -136,6 +131,101 @@ public static void SwitchVersion(AvailablePackage package, string version)
 				Dispatcher.Dispatch(ActionTypes.SWITCH_PACKAGE_VERSION_COMPLETE);
 			});
 		}
+
+		public static void SyncProjectEpic()
+		{
+			CompilePackageListStatus((packageListStatus) =>
+			{
+				if (!packageListStatus.IsProjectUpToDate)
+				{
+					var taskChain = new TaskChain();
+
+					foreach (var packageStatus in packageListStatus.Packages)
+					{
+						var version = packageStatus.RequiredVersion;
+						var packageName = packageStatus.PackageName;
+						var gitUrl = packageStatus.GitUrl;
+						var relativeInstallDirectory = Settings.RelativePackagesDirectoryPath + packageName;
+						var absoluteInstallDirectory = Settings.AbsolutePackagesDirectoryPath + packageName;
+
+						if (packageStatus.IsMissing)
+						{
+							taskChain.Add(
+								new AddSubmoduleTask(gitUrl, relativeInstallDirectory));
+							taskChain.Add(
+								new CheckoutSubmoduleTask(absoluteInstallDirectory, version));
+						}
+						else if (packageStatus.IsOnWrongVersion)
+						{
+							taskChain.Add(
+								new CheckoutSubmoduleTask(
+									absoluteInstallDirectory,
+									version));
+						}
+					}
+
+					taskChain.Execute(() =>
+					{
+						Debug.Log("Synced Project.");
+					});
+				}
+				else
+				{
+					Debug.Log("Project is already up to date.");
+				}
+			});
+		}
+
+		private static void CompilePackageListStatus(Action<PackageListStatus> onComplete)
+		{
+			Dispatcher.Dispatch(ActionTypes.COMPILE_PACKAGE_LIST_STATUS_STARTED);
+
+			var tasks = new TaskChain();
+			var packageListStatus = new PackageListStatus();
+
+			foreach (var package in RootStore.Instance.Packages.InstalledPackages.Value.Packages)
+			{
+				var packageStatus = new PackageStatus
+				{
+					RequiredVersion = package.Version,
+					PackageName = package.Name,
+					GitUrl = package.GitUrl
+				};
+				packageListStatus.Packages.Add(packageStatus);
+				var packageDirectory = Settings.AbsolutePackagesDirectoryPath + package.Name;
+				if (!Directory.Exists(packageDirectory))
+				{
+					packageStatus.IsMissing = true;
+					continue;
+				}
+
+				var task = new GetGitBranchOrTagTask(package);
+				var packageVersion = package.Version;
+				tasks.Add(task);
+				tasks.Add(() =>
+				{
+					packageStatus.IsOnWrongVersion = task.Result != packageVersion;
+				});
+			}
+
+			tasks.Execute(() =>
+			{
+				Debug.Log("Project package status: ");
+
+				foreach (var package in packageListStatus.Packages)
+				{
+					Debug.Log(package);
+				}
+
+				Debug.Log("The project is " +
```

Comment:
Don't think you need the line breaks here, there are lines much longer than this in other classes that don't break.